### ISSUE TITLE
Sort by date added in Downloads panel

### ISSUE THREAD

**Issue by aronj** [Tribler/tribler#1187]
The ability to sort torrents by date added, perhaps presented either by relaxed ISO-8601 date format or "1d1h1m1s ago" format or spoken language "59 minutes ago" / "23 hours ago" / "6 days ago". 
In the case of ISO-8601 timezone could for example be omitted. Maybe even the time could be too much information as densely packed as it will be. I think spoken language could be good for maximizing relevance and readability, optimally with language specific strings. I'm assuming that a value should be visible but that is also to be decided.
Closest entry on [uservoice](http://tribler.uservoice.com/forums/90409-general/suggestions/6862457-sort).

**Comment by synctext** [comment]
Adding correct metadata such as spoken language is a good one, but a bigger ticket.
Doing "date added" should be minor task. But I don't know how accurate the timestamp inside torrents is these days. Should be easy to expose in the GUI. If somebody can work on this as a nice "Tribler introduction exercise" that would be really great. Love to accept a pull-request on this.

[disclaimer]We're understaffed in Tribler. The main developers focus until the summer is getting anonymous downloading+seeding working smoothly plus fixing (severe) performance/usability bugs. So this can take a while...

**Comment by whirm** [comment]
@aronj are you talking about when the torrent was added to the download list or when the torrent file was created?

**Comment by aronj** [comment]
I'm meaning when the torrent was added to the download list, or in other words when it was added to the client. I realize there is room for misunderstandings here. I found the method get_creation_date in TorrentDef. This sounds like something that could be used for this.

**Comment by whirm** [comment]
@aronj thanks for the clarification.

That shouldn't be too dificult to do for a volunteer.

**Comment by alex9311** [comment]
I took a crack at this issue. The get_creation_date method mentioned by @aronj does indeed return the creation date of the torrent file rather than the date it was added to the library. So we don't want that.

As far as I can tell, the date the torrent is added is not currently being stored anywhere. So that would need to be added. I got a bit lost in the code finding out how to add this info to the TorrentDef, but maybe another volunteer can figure it out!
